[Waiting Room] Replaced content from the test waiting room page (#6801)
cloudflare/cloudflare-docs: content/waiting-room/additional-options/test-waiting-room.md

Before you start this tutorial, ensure you have:

- All the [prerequisites](/waiting-room/#prerequisites) completed.
- For this tutorial, we will use an open source tool from apache [jmeter](https://jmeter.apache.org/). You can download the binary from [here](https://jmeter.apache.org/download_jmeter.cgi). 

---

## 1. Download sample script

First, download the [sample](https://github.com/yj7o5/cf-waiting-room-testing/blob/main/plan.jmx) jmeter plan (configuration file) from GitHub.

This sample plan simulates 200 active users visiting the site, slowly ramping up traffic within the first minute and then maintaining 200 active users for the next three minutes. The test plan for this tutorial follows the setup outlined in the next steps.

## 2. Run sample plan

In order to run our test plan click on the **play** button to get the test started which should take roughly around 3-4 minutes.

![Navigation bar](/waiting-room/static/navigation.png)

- Each simulated user has the following attributes:

  - Contains a Cookie jar for cookies persistence

>   - Repeats for 20 times

Reviewer: Is `Cookie jar` the name of something? Is it related to JMter? If it's not a name I don't think we need to capitalize it.

>     - Makes a request to the origin site with waiting room enabled
>     - Logs request details
>     - Pauses for 10 seconds before refreshing the page to make another request to the origin site

Reviewer: ```suggestion
    - Makes a request to the origin site with waiting room enabled.
    - Logs request details.
    - Pauses for 10 seconds before refreshing the page to make another request to the origin site.
```

![User attributes](/waiting-room/static/user-attributes.png)

> Per the plan above, each [Thread Group](https://jmeter.apache.org/usermanual/test_plan.html#thread_group) performs the above action once and the user traffic ramps up within the first minute and keeps a sustained traffic for the next three minutes before users leave the site. You can send more or less traffic than what is being sent in this example by updating these properties.

Reviewer: ```suggestion
Per the plan above, each [Thread Group](https://jmeter.apache.org/usermanual/test_plan.html#thread_group) performs the above action once. The user traffic ramps up within the first minute and keeps a sustained traffic for the next three minutes before users leave the site. You can send more or less traffic than what is being sent in this example by updating these properties.
```

![Visualizing number of threads](/waiting-room/static/threads.png)

## 3. Analyze results

> To analyze the results of your test, you can query Waiting Room Analytics (Beta) via Cloudflare’s GraphQL API to see Total Active Users and Queued Users for each minute of your load test. 

Reviewer: ```suggestion
To analyze the results of your test, you can query Waiting Room Analytics (Beta) via Cloudflare’s GraphQL API to check Total Active Users and Queued Users for each minute of your load test. 
```

}' | tr -d '\n' | curl \
  -X POST \
```

</div>
</details>

From our test, we see the following results (these are extracted from results of the query for readability):

- 15:35:00 UTC              
  - `"totalActiveUsers": 137,`
  - `"totalActiveUsersConfig": 300,`
  - `"totalQueuedUsers": 0`

- 15:36:00 UTC
  - `"totalActiveUsers": 200,`
  - `"totalActiveUsersConfig": 300,`
  - `"totalQueuedUsers": 0`

- 15:37:00 UTC
  - `"totalActiveUsers": 200,`
  - `"totalActiveUsersConfig": 300,`
  - `"totalQueuedUsers": 0`

- 15:38:00 UTC
  - `"totalActiveUsers": 200,`
  - `"totalActiveUsersConfig": 300,`
  - `"totalQueuedUsers": 0`

> The first minute mark, 15:35:00 UTC, shows 137 active users past the waiting room. This is because our traffic was set to gradually ramp up within the first minute and the test did not start exactly at the minute mark. When data was aggregated for the following minute, 15:36:00 UTC, the waiting room reported the total 200 users active we expected to see on the site as each “user” made subrequests. The active user count remained stable at 200 as long as it received subrequests from the traffic sent by the load test.

Reviewer: ```suggestion
The first minute mark, 15:35:00 UTC, shows 137 active users past the waiting room. This is because our traffic was set to gradually ramp up within the first minute and the test did not start exactly at the minute mark. When data was aggregated for the following minute, 15:36:00 UTC, the waiting room reported the total 200 users active we expected on the site as each “user” made subrequests. The active user count remained stable at 200 as long as it received subrequests from the traffic sent by the load test.
```

> {{<Aside type="note" header="Note:">}}

Reviewer: ```suggestion
{{<Aside type="note" header="Note">}}
```

> Please obtain your API Token from the dashboard. Make sure your API Token grants access to the “Analytics” resource. For more on how to get the API Token please follow the guide [here](/analytics/graphql-api/getting-started/authentication/api-token-auth/).

Reviewer: ```suggestion
Obtain your API token from the dashboard. Make sure your API token grants access to the **Analytics** resource. For more information on how to get the API token, follow the [Configure Analytics API token](/analytics/graphql-api/getting-started/authentication/api-token-auth/) guide.
```
